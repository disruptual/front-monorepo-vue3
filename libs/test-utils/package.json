{
  "name": "@dsp/test-utils",
  "version": "0.0.0",
  "dependencies": {
    "@dsp/default-theme": "*",
    "@dsp/ui": "*",
    "axios-mock-adapter": "^1.20.0",
    "fast-glob": "^3.2.11"
  }
}
