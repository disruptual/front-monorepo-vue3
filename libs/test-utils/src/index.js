import { mount } from '@vue/test-utils';
import MockAdapter from 'axios-mock-adapter';

import { plugin as uiPlugin } from '../../ui/src/plugin';
import theme from '../../default-theme/src/index';
import { createQueryClient } from '../../core/src/factories/queryClient.factory';
import { HttpService } from '../../core/src/services/http.service';

export const mountComponent = (component, { props, slots } = {}) => {
  const httpService = new HttpService({ baseURL: '/' });
  const mock = new MockAdapter(httpService.axios);
  mock.onGet('test-file-path').reply(200, '<svg></svg>');

  const wrapper = mount(component, {
    props,
    slots,
    global: {
      plugins: [uiPlugin(theme)],
      provide: {
        HTTP_SERVICE: httpService,
        APP_CONTEXT: {},
        VUE_QUERY_CLIENT: createQueryClient()
      }
    }
  });

  wrapper.httpMock = mock;

  return wrapper;
};
