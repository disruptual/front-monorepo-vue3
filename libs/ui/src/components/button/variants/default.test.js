import { mount } from '@vue/test-utils';
import DspButton from './default.variant.vue';

describe('DspButtonDefault', () => {
  it('should render without throwing', () => {
    expect(() => mount(DspButton)).not.toThrow();
  });

  it('should render correctly', () => {
    expect(mount(DspButton).html()).toMatchSnapshot();
  });

  describe('props', () => {
    it('should render the correct color scheme', () => {
      const { vm } = mount(DspButton, {
        props: {
          scheme: { normal: 500, hover: 600, focus: 700 },
          colorScheme: 'red'
        }
      });

      expect(vm.colors.normal).toBe('var(--color-red-500)');
      expect(vm.colors.hover).toBe('var(--color-red-600)');
      expect(vm.colors.focus).toBe('var(--color-red-700)');
    });

    it('should render a rounded button', () => {
      const wrapper = mount(DspButton, { props: { isRounded: true } });

      const classes = wrapper.find('.dsp-button').classes();

      expect(classes).toContain('dsp-button--is-rounded');
    });

    it('should render a small button', () => {
      const wrapper = mount(DspButton, { props: { size: 'sm' } });

      const classes = wrapper.find('.dsp-button').classes();

      expect(classes).toContain('dsp-button--sm');
    });

    it('should render a medium sized button by default', () => {
      const wrapper = mount(DspButton);

      const classes = wrapper.find('.dsp-button').classes();

      expect(classes).toContain('dsp-button--md');
    });

    it('should render a small button', () => {
      const wrapper = mount(DspButton, { props: { size: 'sm' } });

      const classes = wrapper.find('.dsp-button').classes();

      expect(classes).toContain('dsp-button--sm');
    });

    it('should render a full width button', () => {
      const wrapper = mount(DspButton, { props: { isFullWidth: true } });

      const classes = wrapper.find('.dsp-button').classes();

      expect(classes).toContain('dsp-button--is-fullwidth');
    });

    it('should render an outlined button', () => {
      const wrapper = mount(DspButton, { props: { isOutlined: true } });

      const classes = wrapper.find('.dsp-button').classes();

      expect(classes).toContain('dsp-button--is-outlined');
    });

    it('should render an icon to the left side', () => {
      const wrapper = mount(DspButton, { props: { leftIcon: 'add' } });

      expect(wrapper.find('.left-icon')).toBeDefined();
    });

    it('should render an icon to the right side', () => {
      const wrapper = mount(DspButton, { props: { rightIcon: 'add' } });

      expect(wrapper.find('.right-icon')).toBeDefined();
    });
  });

  describe('events', () => {
    it('should emit click event when the button is clicked', () => {
      const wrapper = mount(DspButton);

      wrapper.find('button').trigger('click');

      expect(wrapper.emitted()).toHaveProperty('click');
    });
  });

  describe('slots', () => {
    it('should render default slot', () => {
      const text = 'Text Slot';
      const wrapper = mount(DspButton, {
        slots: { default: text }
      });
      const button = wrapper.find('.dsp-button');

      expect(button.text()).toContain(text);
    });

    it('should render left icon slot', () => {
      const slot = '<div class="my-left-icon"></div>';
      const wrapper = mount(DspButton, {
        slots: { leftIcon: slot }
      });

      expect(wrapper.find('.my-left-icon')).toBeDefined();
    });

    it('should render right icon slot', () => {
      const slot = '<div class="my-right-icon"></div>';
      const wrapper = mount(DspButton, {
        slots: { rightIcon: slot }
      });

      expect(wrapper.find('.my-right-icon')).toBeDefined();
    });
  });
});
