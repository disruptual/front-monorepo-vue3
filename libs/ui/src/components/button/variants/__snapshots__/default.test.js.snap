// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DspButtonDefault should render correctly 1`] = `
"<button class=\\"dsp-button dsp-button--md\\">
  <!--v-if-->
  <!--v-if-->
</button>"
`;
